app/apollo/models/user.js

```diff
@@ -15,11 +15,75 @@
  */
 
 const mongoose = require('mongoose');
+const jwt = require('jsonwebtoken');
+const { AuthenticationError } = require('apollo-server');
 
 const { AUTH_MODEL } = require('./const');
 const UserSchema = require(`./user.${AUTH_MODEL}.schema`);
 const _ = require('lodash');
 
+const loadMeFromUserToken = async function(userToken, models){
+  let obj, userId, orgId;
+  try {
+    obj = jwt.decode(userToken);
+    userId = obj.userId;
+    orgId = obj.orgId;
+  }catch(err){
+    throw new AuthenticationError('Failed to parse userToken');
+  }
+  if(!userId){
+    throw new AuthenticationError('No user id found in userToken');
+  }
+  const user = await this.findById(userId, {}, { lean:true });
+  if(!user){
+    throw new AuthenticationError('No user found for userToken');
+  }
+  const org = await models.Organization.findById(orgId);
+  if(!org){
+    throw new AuthenticationError('No org found for userToken');
+  }
+  const hasVerifiedToken = _.some(org.orgKeys, (orgKey)=>{
+    try{
+      jwt.verify(userToken, orgKey);
```

Comment:
for the `secretKey`, could we let it = orgKey + some static secret ? That way even some users of the org may know orgKey, they are not be able to sign and pass verify for userToken.